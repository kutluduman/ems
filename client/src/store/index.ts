import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;