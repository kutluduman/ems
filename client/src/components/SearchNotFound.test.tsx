import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import { store } from "../store";
import SearchNotFound from "./SearchNotFound";

describe("SearchNotFound Component", () => {
  test("renders the SearchNotFound Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <SearchNotFound searchQuery="Search Query" />
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});