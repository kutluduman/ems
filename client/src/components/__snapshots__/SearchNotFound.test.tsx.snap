// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchNotFound Component renders the SearchNotFound Component 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter MuiTypography-gutterBottom css-kxw4um-MuiTypography-root"
  >
    Not found
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body2 MuiTypography-alignCenter css-1ypfif-MuiTypography-root"
  >
    No results found for Â 
    <strong>
      "
      Search Query
      "
    </strong>
    . Try checking for typos or using complete words.
  </p>
</div>
`;
